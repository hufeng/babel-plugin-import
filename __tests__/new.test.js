const babel = require('babel-core');
const ImportResolve = require('../lib').default;

describe('new varaible', () => {
  it('test new', () => {
    const example = `
    import {Loading, noop, Star} from 'uikit'
    const l = new Loading();
    console.log(noop());
    console.log(Star);
    `;
    const { code } = babel.transform(example, {
      plugins: [
        [
          ImportResolve,
          {
            libraryName: 'uikit',
            libraryDirectory: '',
            moduleResolver: {
              Loading: {
                path: 'Loading',
                type: 'default'
              },
              noop: {
                path: 'noop',
                type: 'default'
              },
              Star: {
                path: 'star',
                type: '*'
              }
            }
          }
        ]
      ]
    });
    expect(code).toMatchSnapshot();
  });
});
