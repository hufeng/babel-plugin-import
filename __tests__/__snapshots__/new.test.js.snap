// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new varaible test new 1`] = `
"import * as _ from 'uikit/star';
import _default2 from 'uikit/noop';
import _default from 'uikit/Loading';

const l = new _default();
console.log(_default2());
console.log(_);"
`;
